File: virtual-world-simulator/src/Entity/Animals/Sheep.cpp
#include "Sheep.h"

Sheep::Sheep(World* world, Point position) : 
    Animal{ AssetManager::getAssetManager()->getAsset("sheep"), world, position, 'S', 4, 4 }
{
}

bool Sheep::collision(std::fstream& logFile, Entity& entity)
{
	Sheep* sheep = dynamic_cast<Sheep*>(&entity);
	if (sheep != nullptr)
	{
		this->breed<Sheep>(logFile, *sheep);
		return false;
	}

	return Animal::collision(logFile, entity);
}

File: Java/virtual-world-simulator/src/Entities/Animals/Sheep.java
package Entities.Animals;

import App.World;
import Common.Point;
import Entities.Entity;

import javax.swing.*;

public class Sheep extends Animal
{
    public Sheep(World world, Point position)
    {
        super(new ImageIcon("assets/sheep.png"), world, position, 'S', 4, 4);
    }

    public boolean Collision(Entity entity)
    {
        if(entity instanceof Sheep)
        {
            Sheep sheep = new Sheep(this.world, new Point(-1, -1));
            AddChild(sheep);
            return false;
        }

        return super.Collision(entity);
    }
}
